src/backend/libpq/rangerrest.c

```diff
@@ -245,279 +257,326 @@ json_object *create_ranger_request_json_batch(List *args)
+	}
+
+	/* we're done with libcurl, so clean it up */
+	curl_global_cleanup();
+	return ret;
 }
 
 /*
  * arg_list: List of RangerRequestJsonArgs
  */
 int check_privilege_from_ranger_batch(List *arg_list)
 {
-  json_object* jrequest = create_ranger_request_json_batch(arg_list);
-  Assert(jrequest != NULL);
-  const char *request = json_object_to_json_string(jrequest);
-  elog(LOG, "Send JSON request to Ranger: %s", request);
-  Assert(request != NULL);
-  struct curl_context_t curl_context;
-  memset(&curl_context, 0, sizeof(struct curl_context_t));
-
-  /* call GET method to send request*/
-  call_ranger_rest(&curl_context, request);
-  
-  /* free the JSON object */
-  json_object_put(jrequest);
-  
-  /* parse the JSON-format result */
-  RangerACLResult ret = parse_ranger_response(curl_context.response.buffer);
-  
-  /* free response buffer */
-  if (curl_context.response.buffer != NULL)
-  {
-    pfree(curl_context.response.buffer);
-  }
-
-  return ret;
+	json_object* jrequest = create_ranger_request_json_batch(arg_list);
+	Assert(jrequest != NULL);
+	const char *request = json_object_to_json_string(jrequest);
+	elog(LOG, "Send JSON request to Ranger: %s", request);
+	Assert(request != NULL);
+	struct curl_context_t curl_context;
+	memset(&curl_context, 0, sizeof(struct curl_context_t));
+
+	/* call GET method to send request*/
+	if (call_ranger_rest(&curl_context, request) < 0)
+	{
+		return RANGERCHECK_UNKNOWN;
+	}
+
+	/* free the JSON object */
+	json_object_put(jrequest);
+
+	/* parse the JSON-format result */
+	RangerACLResult ret = parse_ranger_response(curl_context.response.buffer);
+	/* free response buffer */
+	if (curl_context.response.buffer != NULL)
+	{
+		pfree(curl_context.response.buffer);
+	}
+
+	return ret;
 }
 
 /*
  * Check the privilege from Ranger for one role
  */
 int check_privilege_from_ranger(char* user, AclObjectKind kind, char* object,
-        List* actions, bool isAll)
+	List* actions, bool isAll)
 {
-    json_object* jrequest = create_ranger_request_json(user, kind, object,
-                                                       actions, isAll);
+	json_object* jrequest = create_ranger_request_json(user, kind, object,
+		actions, isAll);
 
-    Assert(jrequest != NULL);
-    const char* request = json_object_to_json_string(jrequest);
-    elog(LOG, "send JSON request to Ranger: %s", request);
-    Assert(request != NULL);
+	Assert(jrequest != NULL);
+	const char* request = json_object_to_json_string(jrequest);
+	elog(LOG, "send JSON request to Ranger: %s", request);
+	Assert(request != NULL);
 
-    struct curl_context_t curl_context;
-    memset(&curl_context, 0, sizeof(struct curl_context_t));
+	struct curl_context_t curl_context;
+	memset(&curl_context, 0, sizeof(struct curl_context_t));
 
-    /* call GET method to send request*/
-    call_ranger_rest(&curl_context, request);
+	/* call GET method to send request*/
+	if (call_ranger_rest(&curl_context, request) < 0)
+	{
+		return RANGERCHECK_UNKNOWN;
```

Comment:
Here return RANGERCHECK_UNKNOWN has some problem.
check_privilege_from_ranger is call by pg_rangercheck which return AclResult. But RANGERCHECK_UNKNOWN is belong to RangerACLResult. We need to unify them.